app/qan/reporter.go

```diff
@@ -91,11 +92,14 @@ func SparklineData(qr *Reporter, endTs int64, intervalTs int64, queryClassId uin
 	var i int64
 	for i = 0; i < amountOfPoints; i++ {
 		ts := endTs - i*intervalTs
-		if val, ok := queryLogArrRaw[ts]; ok {
-			queryLogArr = append(queryLogArr, val)
-		} else {
-			queryLogArr = append(queryLogArr, qp.QueryLog{uint(i), time.Unix(ts, 0), 0, 0, 0})
+		val, ok := queryLogArrRaw[ts]
+		if !ok {
+			val = qp.QueryLog{
+				Point:    uint(i),
+				Start_ts: time.Unix(ts, 0).UTC(),
```

Comment:
This code was prone to timezone configuration on server, now we always use UTC.

Also code reorganization as `queryLogArr = append(queryLogArr, val)` was used twice